worker/handle-message.ts

```diff
@@ -290,20 +314,20 @@ export const handleFolderModified = (modified: Array<Partial<UserFolder>>): void
 					const oldParent = folders[oldParentId];
 					if (oldParent) {
 						if (!val.l) {
-							oldParent.children = oldParent.children.map((f) => (f.id !== val.id ? f : folder));
+							oldParent.children = oldParent.children.filter((f) => f.id !== val.id);
```

Comment:
Why did you modify this line?

Reply:
in Mails this code does not work in the case in which f.id === val.id. 
when this condition is met the oldParent.children array becomes [folder] at line 317, and at following line we push folder to the array, with the final result being [folder, folder]

in mails the implementation is working correctly, is there maybe a case in calendars where I missed the need for children to be [folder, folder]?